service/controller/internal/nodecontroller/resource/podcondition/create.go

```diff
@@ -0,0 +1,66 @@
+package podcondition
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/kvm-operator/service/controller/key"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	workloadNode, err := key.ToNode(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	var managementPod corev1.Pod
+	err = r.managementK8sClient.Get(ctx, key.NodePodObjectKey(r.cluster, workloadNode), &managementPod)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	condition, shouldUpdate := calculatePodNodeCondition(workloadNode, managementPod)
+	if !shouldUpdate {
+		return nil
+	}
+
+	r.logger.Debugf(ctx, "patching pod node status condition to %#v", condition)
+	err = r.managementK8sClient.Status().Patch(ctx, &managementPod, podConditionPatch{PodCondition: condition})
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	return nil
+}
+
+func calculatePodNodeCondition(node corev1.Node, pod corev1.Pod) (corev1.PodCondition, bool) {
+	nodeReadyCondition, nodeReadyFound := key.FindNodeCondition(node, corev1.NodeReady)
+	podReadyCondition, podReadyFound := key.FindPodCondition(pod, key.WorkloadClusterNodeReady)
+
+	condition := podReadyCondition
+	condition.Type = key.WorkloadClusterNodeReady // In case FindPodCondition returned an empty object
```

Comment:
Currently you're always overriding the value, maybe add a check for `podReadyFound` around this?